### ISSUE TITLE
-Wpointer-sign in arch/s390/include/asm/timex.h

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#861]
```
In file included from ../arch/s390/boot/startup.c:3:                                                                                                            
In file included from ../include/linux/elf.h:5:                                                                                                                 
In file included from ../arch/s390/include/asm/elf.h:132:                                                                                                       
In file included from ../include/linux/compat.h:10:                                                                                                             
In file included from ../include/linux/time.h:74:                                                                                                               
In file included from ../include/linux/time32.h:13:                                                                                                             
In file included from ../include/linux/timex.h:65:                                                                                                              
../arch/s390/include/asm/timex.h:160:20: warning: passing 'unsigned char [16]' to parameter of type 'char *' converts between pointers to integer types with different sign [-Wpointer-sign]                                                                                                                                    
        get_tod_clock_ext(clk);                                                                                                                                 
                          ^~~                                                                                                                                   
../arch/s390/include/asm/timex.h:149:44: note: passing argument to parameter 'clk' here                                                                         
static inline void get_tod_clock_ext(char *clk)                                                                                                                 
                                           ^
```

Should just need to change the type.

**Comment by nathanchance** [comment]
Patch submitted: https://lore.kernel.org/lkml/20200208140858.47970-1-natechancellor@gmail.com/

**Comment by nickdesaulniers** [comment]
accepted: https://git.kernel.org/pub/scm/linux/kernel/git/s390/linux.git/commit/?h=fixes&id=0f8a206df7c920150d2aa45574fba0ab7ff6be4f

**Comment by nathanchance** [comment]
Mainline: https://git.kernel.org/torvalds/c/0f8a206df7c920150d2aa45574fba0ab7ff6be4f
